subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -889,6 +891,252 @@ static int nrf_cloud_encode_service_info_ui(const struct nrf_cloud_svc_info_ui *
+		if (ret) {
+			return ret;
+		}
+	}
+
+	return ret;
+}
+
+static int modem_info_network_encode(struct network_param *network, cJSON *json_obj)
+{
+	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE];
+	int ret;
+
+	static const char lte_string[]	 = "LTE-M";
+	static const char nbiot_string[] = "NB-IoT";
+	static const char gps_string[]	 = " GPS";
+
+	__ASSERT_NO_MSG(network != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	ret = modem_info_add_data(&network->current_band, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_add_data(&network->sup_band, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_add_data(&network->area_code, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_add_data(&network->current_operator, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_add_data(&network->ip_address, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_add_data(&network->ue_mode, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_name_get(network->cellid_hex.type, data_name);
+	if (ret < 0) {
+		return ret;
+	}
+	data_name[ret] = '\0';
+
+	ret = json_add_num_cs(json_obj, data_name, network->cellid_dec);
+	if (ret) {
+		return ret;
+	}
+
+	network->network_mode[0] = '\0';
+
+	if (network->lte_mode.value == 1) {
+		strcat(network->network_mode, lte_string);
+	} else if (network->nbiot_mode.value == 1) {
+		strcat(network->network_mode, nbiot_string);
+	}
+
+	if (network->gps_mode.value == 1) {
+		strcat(network->network_mode, gps_string);
+	}
+
+	ret = json_add_str_cs(json_obj, "networkMode", network->network_mode);
+
+	return ret;
+}
+
+static int modem_info_sim_encode(struct sim_param *sim, cJSON *json_obj)
+{
+	int ret;
+
+	__ASSERT_NO_MSG(sim != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	ret = modem_info_add_data(&sim->uicc, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	if (IS_ENABLED(CONFIG_MODEM_INFO_ADD_SIM_ICCID)) {
+		ret = modem_info_add_data(&sim->iccid, json_obj);
+		if (ret) {
+			return ret;
+		}
+	}
+
+	if (IS_ENABLED(CONFIG_MODEM_INFO_ADD_SIM_IMSI)) {
+		ret = modem_info_add_data(&sim->imsi, json_obj);
+	}
+
+	return ret;
+}
+
+static int modem_info_device_encode(struct device_param *device, cJSON *json_obj)
+{
+	int ret;
+
+	__ASSERT_NO_MSG(device != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	ret = modem_info_add_data(&device->modem_fw, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_add_data(&device->battery, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_add_data(&device->imei, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = json_add_str_cs(json_obj, "board", device->board);
+	if (ret) {
+		return ret;
+	}
+
+	ret = json_add_str_cs(json_obj, "appVersion", device->app_version);
+	if (ret) {
+		return ret;
+	}
+
+	ret = json_add_str_cs(json_obj, "appName", device->app_name);
+	if (ret) {
+		return ret;
+	}
+
+	return ret;
+}
+
+static int modem_info_json_object_encode(struct modem_param_info *modem,
```

Comment:
Please change the function prefix here as well.

Reply:
✔